components/integrations/abstract_incoming_webhook.tsx

```diff
@@ -1,87 +1,100 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
-import {FormattedMessage} from 'react-intl';
+import {FormattedMessage, MessageDescriptor} from 'react-intl';
 import {Link} from 'react-router-dom';
 
 import BackstageHeader from 'components/backstage/components/backstage_header';
 import ChannelSelect from 'components/channel_select';
 import FormError from 'components/form_error';
 import SpinnerButton from 'components/spinner_button';
+import {Team} from '@mattermost/types/teams';
 import {localizeMessage} from 'utils/utils';
+import {IncomingWebhook} from '@mattermost/types/integrations';
 
-export default class AbstractIncomingWebhook extends React.PureComponent {
-    static propTypes = {
+interface State {
+    displayName: string;
+    description: string;
+    channelId: string;
+    channelLocked: boolean;
+    username: string;
+    iconURL: string;
+    saving: boolean;
+    serverError: string;
+    clientError: JSX.Element | null;
+}
 
-        /**
-        * The current team
-        */
-        team: PropTypes.object.isRequired,
+interface Props {
 
-        /**
-        * The header text to render, has id and defaultMessage
-        */
-        header: PropTypes.object.isRequired,
+    /**
+    * The current team
+    */
+    team: Team;
 
-        /**
-        * The footer text to render, has id and defaultMessage
-        */
-        footer: PropTypes.object.isRequired,
+    /**
+    * The header text to render, has id and defaultMessage
+    */
+    header: MessageDescriptor;
 
-        /**
-        * The spinner loading text to render, has id and defaultMessage
-        */
-        loading: PropTypes.object.isRequired,
+    /**
+    * The footer text to render, has id and defaultMessage
+    */
+    footer: MessageDescriptor;
 
-        /**
-        * The server error text after a failed action
-        */
-        serverError: PropTypes.string.isRequired,
+    /**
+    * The spinner loading text to render, has id and defaultMessage
+    */
+    loading: MessageDescriptor;
 
-        /**
-        * The hook used to set the initial state
-        */
-        initialHook: PropTypes.object,
+    /**
+    * The server error text after a failed action
+    */
+    serverError: string;
 
-        /**
-        * Whether to allow configuration of the default post username.
-        */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
+    /**
+    * The hook used to set the initial state
+    */
+    initialHook?: IncomingWebhook | Record<string, never>;
```

Comment:
why do we use never here?

Reply:
Because the initialHook value can be {}. Record<string, never> is an empty object in the sense that you can't add any string-type accessor properties to it, so it will always be empty.